src/app/modules/rup/components/ejecucion/dacion-turno/asignar-turno.ts

```diff
@@ -184,6 +191,48 @@ export class DinamicaFormComponent implements OnInit {
                 }
 
             });
+        } else {
+            if (!this.horaTurno) {
+                return this.plex.info('warning', 'Debe seleccionar hora del sobreturno');
+            }
+
+            const patch = {
+                'op': 'agregarSobreturno',
+                'sobreturno': {
+                    horaInicio: this.combinarFechas(this.agenda.horaInicio, this.horaTurno),
+                    estado: 'asignado',
+                    tipoPrestacion: this.turnoTipoPrestacion,
+                    paciente: this.pacienteActivo,
+                }
+            };
+
+            this.serviceAgenda.patch(this.agenda.id, patch).subscribe(agenda => {
+                const sobreTurno = agenda.sobreturnos[agenda.sobreturnos.length - 1];
+                if (sobreTurno) {
+                    return this.servicioPrestacion.crearPrestacion(paciente, sobreTurno.tipoPrestacion, 'ejecucion', sobreTurno.horaInicio, sobreTurno.id).subscribe(
+                        prestacion => {
+                            this.router.navigate(['rup/ejecucion/', prestacion.id]);
+                        }
+                    );
+                } else {
+                    this.plex.info('danger', 'No fue posible crear la prestación');
+                    return EMPTY;
+                }
+
+            });
```

Comment:
Recordemos que no es buena practica anidar subscribes. Se sugiere el siguiente codigo:
```suggestion
this.serviceAgenda.patch(this.agenda.id, patch).pipe(
                switchMap(agenda => {
                    const sobreTurno = agenda.sobreturnos[agenda.sobreturnos.length - 1];
                    if (sobreTurno) {
                        return this.servicioPrestacion.crearPrestacion(paciente, sobreTurno.tipoPrestacion, 'ejecucion', sobreTurno.horaInicio, sobreTurno.id);
                    } else {
                        this.plex.info('danger', 'No fue posible crear la prestación');
                        return EMPTY;
                    }
                }),
                tap(prestacion => {
                    if (prestacion) {
                        this.router.navigate(['rup/ejecucion/', prestacion.id]);
                    }
                })
            );
```

Reply:
esto en que quedo??